osbs/tekton.py

```diff
@@ -380,6 +380,11 @@ def start_pipeline_run(self):
             raise OsbsException("No input data provided for pipeline run to start")
 
         run_name = self.input_data.get('metadata', {}).get('name')
+        labels = self.input_data.get('metadata', {}).get('labels', {})
+
+        if labels:
```

Comment:
This `if` looks imply two things, `metadata` might be omitted, if no, then `metadata.labels` might be omitted. However, the PipelineRun definition is read from predefined yaml file, and in function `_set_binary_container_pipeline_data`, the metadata  is accessed directly:

```
pipeline_run_data['metadata']['name'] = pipeline_run_name
```

If the `pipeline_run_data` should always has key `metadata`, then it is unnecessary to get metadata by using `dict.get`, `self.input_data.get('metadata', {})`.

Meanwhile, IMO, `start_pipeline_run` is not a good place to prepare the labels, it should just start the pipeline run as the name implies. `_set_binary_container_pipeline_data` prepares the pipeline run data including labels, there should be the right place to sanitize the lable values.

Reply:
+1 to sanitize labels in the `_set_*_pipeline_data` methods instead

Reply:
about first note, this method should be general one, not JUST the one which is used by _set* methods

Reply:
I prefer to handle it in tekton module, since it is ocp thing, I could make helper method here and call that if you want, but I want to keep it here

Reply:
That's a good point. I would argue that those api methods already have to deal with some Tekton/OCP details that they maybe shouldn't, but I agree that the tekton module is a better place to sanitize labels